tests/components/unifi/test_controller.py

```diff
@@ -279,6 +286,48 @@ async def test_controller_unknown_error(hass):
     assert hass.data[UNIFI_DOMAIN] == {}
 
 
+async def test_config_entry_updated(hass, aioclient_mock):
+    """Calling reset when the entry has been setup."""
+    config_entry = await setup_unifi_integration(hass, aioclient_mock)
+    controller = hass.data[UNIFI_DOMAIN][config_entry.entry_id]
+
+    event_call = Mock()
+    unsub = async_dispatcher_connect(hass, controller.signal_options_update, event_call)
+
+    hass.config_entries.async_update_entry(
+        config_entry, options={CONF_TRACK_CLIENTS: False, CONF_TRACK_DEVICES: False}
+    )
+    await hass.async_block_till_done()
+
+    assert controller.option_track_clients is False
+    assert controller.option_track_devices is False
+
+    event_call.assert_called_once()
+
+    unsub()
+
+
+async def test_config_entry_updated_no_matching_entry(hass, aioclient_mock):
+    """Verify options aren't updated if config entry not available."""
+    config_entry = await setup_unifi_integration(hass, aioclient_mock)
+    controller = hass.data[UNIFI_DOMAIN].pop(config_entry.entry_id)
+
+    event_call = Mock()
+    unsub = async_dispatcher_connect(hass, controller.signal_options_update, event_call)
+
+    hass.config_entries.async_update_entry(
+        config_entry, options={CONF_TRACK_CLIENTS: False, CONF_TRACK_DEVICES: False}
+    )
+    await hass.async_block_till_done()
+
+    assert controller.option_track_clients is True
```

Comment:
These tests are moving in the opposite direction of where we want to go. We want to decrease the interactions with integration details, not increase them.

Reply:
I think this has been improved quite a lot now